includes/class-coblocks-form.php

```diff
@@ -897,6 +904,66 @@ public function process_form_submission( $atts ) {
 
 	}
 
+	/**
+	 * Setup the email subject line, replacing any found shortcodes.
+	 * Note: [email] will be replaced with the value of field-email
+	 *       [name] will be replaced with the value of field-name etc.
+	 *
+	 * @param  array  $atts           Block attributes array.
+	 * @param  string $email_field_id Email field ID.
+	 * @param  string $name_field_id  Nane field ID.
+	 * @return string Email subject.
+	 */
+	private function setup_email_subject( $atts, $email_field_id, $name_field_id ) {
+
+		$default_subject = __( 'Form submission from [email]', 'coblocks' );
+		$subject         = isset( $atts['subject'] ) ? sanitize_text_field( $atts['subject'] ) : $default_subject;
+
+		preg_match_all( '/\[(.*?)\]/i', $subject, $matches );
+
+		if ( isset( $matches[1] ) ) {
+
+			array_walk(
+				$matches[1],
+				function( $match, $key ) use ( $matches, &$subject, &$email_field_id, &$name_field_id ) {
+					$slug_match = strtolower( preg_replace( '/\s+/', '', $match ) );
+
+					// phpcs:disable WordPress.Security.NonceVerification.Missing
+					if ( 'name' === $slug_match ) {
+						if ( isset( $_POST[ $name_field_id ]['value'] ) ) {
+
+							$name_field_value = is_array( $_POST[ $name_field_id ]['value'] ) ? sanitize_text_field( implode( ' ', $_POST[ $name_field_id ]['value'] ) ) : sanitize_text_field( $_POST[ $name_field_id ]['value'] );
+							$value            = ( '' === $name_field_value ) ? $matches[0][ $key ] : $name_field_value;
```

Comment:
We can probably run a check to see if `$name_field_value` is empty instead of an empty string.
```php
$value = empty( $name_field_value ) ? $matches[0][ $key ] : $name_field_value;
```

Reply:
using `empty()` now.